packages/hub/src/hub.ts

```diff
@@ -248,7 +248,7 @@ export class Hub implements HubInterface {
    * @inheritDoc
    */
   public captureEvent(event: Event, hint?: EventHint): string {
-    const eventId = (this._lastEventId = uuid4());
+    const eventId = event.type === 'transaction' ? uuid4() : (this._lastEventId = uuid4());
```

Comment:
I'd write it as:

```suggestion
    const eventId = uuid4();
    
    if (event.type !== 'transaction') {
      this._lastEventId = eventId;
    }
    
```

Slightly easier to read imo.